### DISCUSSION TITLE
Help: Multiple values for a rounded corner? What does it mean?

### DISCUSSION THREAD

**Discussion by vaaski** [Irev-Dev/Round-Anything#38] [Q&A] [answered]
Hey, I am trying to make a magsafe accessory for my iPhone in openscad and I found [this excellent specsheet](https://developer.apple.com/accessories/Accessory-Design-Guidelines.pdf#page=199) of the phone directly from apple, but the rounding of it's corner isn't a trivial single value that I can wrap my head around as a (web-) dev.

> ![image](https://user-images.githubusercontent.com/17879327/152181095-3dec0015-859b-448b-b258-d9fc4d888a94.png)
> there are 6 values for one rounded corner

I've done some digging but can't really pinpoint anything, here are some things that _might_ be related:
- [Superellipses](https://en.wikipedia.org/wiki/Superellipse) - might go into the right direction but doesn't explain multiple values
- [in-depth analasys of apple's squircle icons](https://www.figma.com/blog/desperately-seeking-squircles/) - too technical for me to understand :(
- [unrelated iphone fish fillet case](https://user-images.githubusercontent.com/17879327/152182039-2387757b-6b2b-4ed8-bb5e-f705ba09375f.png) - didn't expect this showing up when researching fillet

**I'd be very grateful if someone could point me in the right direction on how to use multiple values for a rounded corner**

<sub>related: idcrook/psychic-winner#2</sub>

**Comment by Irev-Dev** [comment, marked as answer]
My guess is they're just giving a bunch of coordinates on the curve and it's a spline linking them all up, they clearly show the datum as the top left corner. I would probably just try plugging the coordinates into a spline and use BOSL https://github.com/revarbat/BOSL2

**Reply by vaaski** [reply]
thank you so much! i‘ll look into that!

**Reply by vaaski** [reply]
After finally sinking another couple hours into this today I think i finally solved it!
Your assumption was spot on, but at first I was getting results that were way to round, similar to idcrook/psychic-winner#2

![image](https://user-images.githubusercontent.com/17879327/163422101-1d01cd84-0f88-4071-9466-478ecc646044.png)
red is from the [spec sheet (page 199)](https://developer.apple.com/accessories/Accessory-Design-Guidelines.pdf), converted to svg and imported into openscad for comparison

In the end my mistake was to not include 0 in the array of coordinates, now it's perfectly spot on!

![image](https://user-images.githubusercontent.com/17879327/163422560-3f52cd62-a616-4b23-820c-ccbd5a35b0db.png)

**Reply by Irev-Dev** [reply]
A happy ending! so good 🎉
